File: Puzzles/CodingChallenges/findminmax.r

nums<-c(3,5,1,2,4,8)

find_min_max<-function(number_list){

  list_length=length(number_list)
  local_min=number_list[1]
  local_max=number_list[1]
  
  for(k in 1:list_length){
  
    if(number_list[k]<local_min){
      local_min=nums[k]
    }
    
    else{
      
      if(number_list[k]>local_max){
      local_max=number_list[k]
      }
    }
  }
  
  output=c(local_min,local_max)
}

File: Puzzles/CodingChallenges/findminmax.js
function find_min_max(number_list){

    let local_min=number_list[0];
    let local_max=number_list[0];

    for(var k=0;k<number_list.length;k++){

        if(number_list[k]<local_min){
            local_min=number_list[k];
        }
        else{
            if(number_list[k]>local_max){
                local_max=number_list[k];
            }
        }
    }
    return [local_min,local_max]
}

console.log(find_min_max([1,4,2,7,6]))
